app/views/finance/provider/settings/show.html.slim

```diff
@@ -1,135 +1,114 @@
-
+                      - if is_current
+                        = fields.input name, as: :patternfly_checkbox,
+                                             id: ['account', 'payment', payment_gateway.type, name].join('_'),
+                                             input_html: { checked: @account.payment_gateway_options[name].present? }
+                      - else
+                        = fields.input name, as: :patternfly_checkbox,
+                                             label: label,
+                                             checked: false,
+                                             id: ['account', 'payment', payment_gateway.type, name].join('_')
                     - else
-                      = fields.label name, label
-                      - if enabled
-                        = fields.text_field name, value: @account.payment_gateway_options[name]
+                      - if is_current
+                        = fields.input name, as: :patternfly_input,
+                                             input_html: { value: @account.payment_gateway_options[name] }
                       - else
-                        = fields.text_field name, value: '', disabled: true
-
-      = form.actions do
-        = form.commit_button 'Save changes'
-
-  javascript:
-    $(function() {
-      var form = $("#payment-gateway-form");
-      var gateway = $("#account_payment_gateway_type");
-      var old_val = gateway.val();
-      var new_val, old_val;
-
-      gateway.change(function() {
-        $('.payment_gateway_options:visible').hide(function() {
-          $(this).find('input').attr('disabled', true);
-        });
-
-        $('#payment_gateway_' + $(this).val()).show(function() {
-          $(this).find('input').removeAttr('disabled');
-        });
-      });
-
-      form.submit(function() {
-        var new_val = $("#account_payment_gateway_type").val();
-
-        if(new_val == old_val) { return; }
-
-        var first_question = "Changing your payment gateway will make all your buyers' credit card details invalid, "+
-        "so they will have to enter the credit card information again to be billed next month.\n\n"+
-        "Are you sure you want to change your payment gateway?";
-
-        var second_question = "Are you sure? If you click OK, all credit card details from all your customers will be erased and won't be recoverable.\n"+
-        "You'll need to ask your customers to enter their credit card details again to be able to bill them.\n\n" +
-        "Do you want to remove all credit card details from your customers?";
-
-
-        return confirm(first_question) && confirm(second_question);
-      });
-    });
+                        = fields.input name, as: :patternfly_input,
+                                             disabled: true
```

Comment:
Also, `disabled: true` doesn't seem to be handled by the patternfly input :thinking:

Reply:
Actually, how about this instead of multiple branches:

```
                    - if payment_gateway.boolean_field_keys.include?(name)
                      = fields.input name, as: :patternfly_checkbox,
                              id: ['account', 'payment', payment_gateway.type, name].join('_'),
                              label: label,
                              input_html: { checked: is_current ? @account.payment_gateway_options[name].present? : false, disabled: !is_current }
                    - else
                      = fields.input name, as: :patternfly_input,
                              label: label,
                              input_html: { value: is_current ? @account.payment_gateway_options[name] : '', disabled: !is_current }
```

Except that the JS handing of `disabled` has been removed, so it doesn't work. So, I'd even say that we can remove the `disabled` piece, because the part of the form is hidden anyway, so if it's not visible, why disable it?

Reply:
AFAIK this `disabled` doesn't do anything and it's probably a typo. Moreover, disabling the inputs would prevent the user from changing to another gateway.

NOTE: Current means this is the gateway currently in use, not the one currently selected in the dropdown.

Reply:
Hm, well, I guess initially "current" is for setting "currently in use", but also the toggling enabled/disabled was done for "currently selected" https://github.com/3scale/porta/pull/3596/files#diff-8cc2bd457556805766347846af6a80e7635d681acd9fa5fb0a4cd3c7a05751e3L109-L117

I agree that there is no need to enable/disable the inputs, as they are already visible/hidden.

Reply:
OK I see! 👍🏻  I forgot about that JS callback.